lib/Conversion/TritonGPUToLLVM/ConvertLayoutOpToLLVM/SharedToDotOperandFMA.cpp

```diff
@@ -70,164 +32,229 @@ Value getStructFromValueTable(ArrayRef<Value> vals,
-      for (unsigned mm = 0; mm < sizePerThread; ++mm) {
-        res[{m + mm, k}] = elems[index++];
-      }
-  }
-  return res;
+bool isSwizzled(SharedEncodingAttr layout) { return layout.getMaxPhase() != 1; }
+
+SmallVector<Value> swizzleIndices(ConversionPatternRewriter &rewriter,
+                                  Location loc, SmallVector<Value> rawIndices,
+                                  SharedEncodingAttr layout) {
+  const auto &order = layout.getOrder();
+  auto rank = order.size();
+
+  if (!isSwizzled(layout))
+    return rawIndices;
+
+  auto vec = i32_val(layout.getVec());
+  auto perPhase = i32_val(layout.getPerPhase());
+  auto maxPhase = i32_val(layout.getMaxPhase());
+
+  auto fastIdx = rawIndices[order[0]];
+  auto secondIdx = rawIndices[order[1]];
+  // Original algorithm taken from getSwizzledSharedPtrs function
+  // (TritonGPUToLLVMBase.h)
+  //
+  // phase = (secondIdx // perPhase) % maxPhase
+  // swizzledGroup = ((fastIdx // vec) ^ phase) * vec
+  // groupRemainder = fastIdx % vec
+  // colOff = swizzledGroup + groupRemainder
+  auto phase = urem(udiv(secondIdx, perPhase), maxPhase);
+  auto swizzledGroup = mul(xor_(udiv(fastIdx, vec), phase), vec);
+  auto groupRemainder = urem(fastIdx, vec);
+  auto colOff = add(swizzledGroup, groupRemainder);
+
+  SmallVector<Value> swizzledIndices = rawIndices;
+  swizzledIndices[order[0]] = colOff;
+
+  return swizzledIndices;
 }
 
-Value loadAFMA(Value A, Value llA, BlockedEncodingAttr dLayout, Value thread,
-               Location loc, const LLVMTypeConverter *typeConverter,
-               ConversionPatternRewriter &rewriter) {
-  auto aTensorTy = cast<MemDescType>(A.getType());
-  auto aLayout = cast<SharedEncodingAttr>(aTensorTy.getEncoding());
-  auto aShapePerCTA = getShapePerCTA(aTensorTy);
-
-  auto aOrder = aLayout.getOrder();
-  auto order = dLayout.getOrder();
-
-  bool isARow = aOrder[0] == 1;
-
-  auto aSmem = getSharedMemoryObjectFromStruct(
-      loc, llA, typeConverter->convertType(aTensorTy.getElementType()),
-      rewriter);
-  Value strideAM = aSmem.strides[0];
-  Value strideAK = aSmem.strides[1];
-  Value strideA0 = isARow ? strideAK : strideAM;
-  Value strideA1 = isARow ? strideAM : strideAK;
-  int aNumPtr = 8;
-  int K = aShapePerCTA[1];
-  int M = aShapePerCTA[0];
-
-  auto shapePerCTATile = getShapePerCTATile(dLayout);
-  auto sizePerThread = getSizePerThread(dLayout);
-
-  Value _0 = i32_val(0);
-
-  Value mContig = i32_val(sizePerThread[order[1]]);
-
-  // threadId in blocked layout
-  auto threadIds = getThreadIds(thread, shapePerCTATile, sizePerThread, order,
-                                rewriter, loc);
-  Value threadIdM = threadIds[0];
-
-  Value offA0 = isARow ? _0 : mul(threadIdM, mContig);
-  Value offA1 = isARow ? mul(threadIdM, mContig) : _0;
-  SmallVector<Value> aOff(aNumPtr);
-  for (int i = 0; i < aNumPtr; ++i) {
-    aOff[i] = add(mul(offA0, strideA0), mul(offA1, strideA1));
+/**
```

Comment:
This is not the commonly used style. Can we follow the common style in the codebase to be consistent?

Reply:
I will change it.
Do we have a code style guide for stuff like this?

Sometimes it is not obvious which option to pick, I've seen few different styles of comments in many places in our code base and seems this depends on a code author taste.

Reply:
Yeah it's a bit confusing. I think we should follow MLIR style in general: https://github.com/llvm/llvm-project/blob/main/mlir/include/mlir/IR/Operation.h